### PR TITLE
[279,525] ACRs are considered private (regression)

### LINKED ISSUE #279: [#148 Implement TES Task Runner]Private container registry handling through the runner
Motivation to prioritize this: vmsizes that are gen-2 capable are prone to failures in various different ways due to the fact that batch does not have any gen-2 approved images for workloads using containers.

* Increasing security by no longer placing credentials in plain view in the azure pool configuration by placing tes-runner in full control of handing all phases of the interactions with the container registries.
* Improving reliability for Gen2 VMSizes when running in Azure Batch by enabling the docker settings to be fixed BEFORE they are used.
* Improving performance by reducing the time required to queue a TesTask into Azure Batch, while also reducing network traffic and Azure management API calls.

Note: the implementation does NOT work with customizing ACR domain names.

### LINKED ISSUE #525: Private task images result in ContainerTaskSettingsNotFound error
**Describe the bug**
`TesTask`s using images on authenticated-accessible container registries fail with an `EXECUTOR_ERROR of `ContainerTaskSettingsNotFound` because of a recent change that removed the task container configuration that pairs with the pool container configuration in Azure Batch.

**Steps to Reproduce**
Use a workflow that uses private docker images.

**Expected behavior**
Task does not quickly fail with an error message of "Container-enabled compute node requires task container settings".

**Additional context**
Long term #279 needs to be implemented because vmsizes that are gen-2 capable are prone to failures in various different ways due to the fact that batch does not have any gen-2 approved images for workloads using containers. This fix does not address that issue.

### PR DESCRIPTION
**Pull request by giventocode**
fixes #279 
fixes #525

In this PR:
 - `DockerExecutor` uses the MI token to authenticate the call to pull the image if the container registry is ACR.
 - Pool registry container configuration is removed from TES

### CODE REVIEW

File: src/Tes.Runner.Test/Docker/DockerExecutorTests.cs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        mockCredentialsManager.Setup(c => c.GetTokenCredential(It.IsAny<RuntimeOptions>()))
+            .Returns(mockCredentials.Object);
+        mockCredentials.Setup(c => c.GetTokenAsync(It.IsAny<TokenRequestContext>(), It.IsAny<CancellationToken>()))
+            .ReturnsAsync(accessToken);
+
+        dockerExecutor = new DockerExecutor(mockDockerClient.Object, mockStreamLogReader.Object, mockCredentialsManager.Object);
+    }
+
+    [TestMethod]
+    public async Task RunOnContainerAsync_AzureContainerRegistryImage_AuthInfoIsProvided()
```
Comment (BMurri) [inline src/Tes.Runner.Test/Docker/DockerExecutorTests.cs:63]: This method requires `[TestCategory("RequiresRoot")]`

File: src/Tes.Runner.Test/Docker/DockerExecutorTests.cs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        await dockerExecutor.RunOnContainerAsync(execOptions);
+
+        Assert.IsNotNull(captureAuthConfig);
+        Assert.AreEqual("https://test.azurecr.io", captureAuthConfig!.ServerAddress);
+        Assert.AreEqual(DockerExecutor.ManagedIdentityUserName, captureAuthConfig.Username);
+        Assert.AreEqual(accessToken.Token, captureAuthConfig.Password);
+    }
+
+    [TestMethod]
+    public async Task RunOnContainerAsync_NotAzureContainerRegistryImage_AuthInfoNull()
```
Comment (BMurri) [inline src/Tes.Runner.Test/Docker/DockerExecutorTests.cs:77]: This method requires `[TestCategory("RequiresRoot")]`

File: src/Tes.Runner/Docker/ContainerRegistryAuthorizationManager.cs [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+using Tes.Runner.Authentication;
+using Tes.Runner.Models;
+using Tes.Runner.Transfer;
+
+namespace Tes.Runner.Docker
+{
+    public class ContainerRegistryAuthorizationManager
+    {
+        const string AzureContainerRegistryHostSuffix = ".azurecr.io";
+        public const string NullGuid = "00000000-0000-0000-0000-000000000000";
```
Comment (MattMcL4475) [inline src/Tes.Runner/Docker/ContainerRegistryAuthorizationManager.cs:17]: Could use `Guid.Empty` for improved readability

File: src/Tes.Runner/Docker/ContainerRegistryAuthorizationManager.cs [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+            ArgumentNullException.ThrowIfNull(tokenCredentialsManager);
+
+            this.tokenCredentialsManager = tokenCredentialsManager;
+        }
+
+        public async Task<AuthConfig?> TryGetAuthConfigForAzureContainerRegistryAsync(string imageName, string? imageTag, RuntimeOptions runtimeOptions)
+        {
+            if (!TryParseAzureContainerRegisteryParts(imageName, out var imageParts))
+            {
+                logger.LogInformation(@"The image is not in ACR. Registry: {ImageName}", imageName);
```
Comment (MattMcL4475) [inline src/Tes.Runner/Docker/ContainerRegistryAuthorizationManager.cs:33]: I'd remove this to reduce log noise

**Review by MattMcL4475 — APPROVED** [review]

**Pull request merged**